test/libsolutil/CommonIO.cpp

```diff
@@ -64,6 +64,21 @@ BOOST_AUTO_TEST_CASE(readFileAsString_symlink)
 	BOOST_TEST(readFileAsString((tempDir.path() / "symlink.txt").string()) == "ABC\ndef\n");
 }
 
+BOOST_AUTO_TEST_CASE(readUntilEnd_no_ending_newline)
+{
+	BOOST_TEST(readUntilEnd(istringstream("ABC\ndef")) == "ABC\ndef");
+}
+
+BOOST_AUTO_TEST_CASE(readUntilEnd_with_ending_newline)
+{
+	BOOST_TEST(readUntilEnd(istringstream("ABC\ndef\n")) == "ABC\ndef\n");
+}
+
+BOOST_AUTO_TEST_CASE(readUntilEnd_empty)
+{
+	BOOST_TEST(readUntilEnd(istringstream("")) == "");
+}
+
```

Comment:
Can you also add a test case with Windows line endings? That's the case that would trip up your previous implementation so better have it covered.

Reply:
We should test this. Maybe it's also different depending on the platform?

If newlines were transformed on windows, this should get a changelog entry.

Reply:
I checked and looks like it was not converting newlines after all. It's because the line ending that `getline()` strips is always `\n` and not platform-dependent as I expected.

So the only thing the previous implementation was doing wrong was always adding an extra `\n` at the end of the file. Added a changelog entry saying this.

Reply:
Looks like outputting the `rdbuf()` does advance the stream (the function is `const` but the pointer it returns is not) so I changed the parameter back to a non-`const` reference and renamed the function back to `readUntilEnd()`.